api/types.proto

```diff
@@ -277,6 +277,40 @@ message UpdateConfig {
 
 	// Amount of time between updates.
 	Duration delay = 2 [(gogoproto.nullable) = false];
+
+	enum OnFailure {
+		PAUSE = 0;
+		CONTINUE = 1;
+		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
+		// (#486)
+	}
+
+	// OnFailure is the action to take when an update failures. Currently,
+	// a failure is defined as a single updated task failing to reach the
+	// RUNNING state. In the future, there will be configuration to define
+	// what is treated as a failure (see #486 for a proposal).
+	OnFailure on_failure = 3;
+}
+
+// UpdateStatus is the status of an update in progress.
+message UpdateStatus {
```

Comment:
It might be better to do convergence checks on the client side. It's relatively straightforward to query the service's tasks and check on the progress. Doing it this way would avoid a need to update the service every time one of its tasks is updated.

If we do decide to do it server-side, it probably belongs in a separate message under `Service`.